From 6da6f2613830eb1ebf83235aa9cd388d2a3329c3 Mon Sep 17 00:00:00 2001
From: "yi.liu" <yi.liu@amlogic.com>
Date: Wed, 26 Apr 2023 14:43:43 +0800
Subject: [PATCH] amlbian: amlbian usr mbr partition [1/1]

PD#SWPL-121716

Problem:
amlbian usr mbr partition.

Solution:
amlbian usr mbr partition.

Verify:
local

Signed-off-by: yi.liu <yi.liu@amlogic.com>
Change-Id: Iaebfe92d03316b2735406a2edd454e01f1d90d33
---
 .../amlogic/defconfigs/t7_an400_lpddr4x_debian_defconfig  | 1 +
 drivers/mmc/Kconfig                                       | 8 ++++++++
 drivers/mmc/aml_emmc_partition.c                          | 1 -
 3 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/board/amlogic/defconfigs/t7_an400_lpddr4x_debian_defconfig b/board/amlogic/defconfigs/t7_an400_lpddr4x_debian_defconfig
index 2d2ec491bd..59b3f36c78 100644
--- a/board/amlogic/defconfigs/t7_an400_lpddr4x_debian_defconfig
+++ b/board/amlogic/defconfigs/t7_an400_lpddr4x_debian_defconfig
@@ -129,3 +129,4 @@ CONFIG_LZ4=y
 CONFIG_LZO=y
 CONFIG_ZSTD=y
 CONFIG_OF_LIBFDT_OVERLAY=y
+CONFIG_PTBL_MBR=y
diff --git a/drivers/mmc/Kconfig b/drivers/mmc/Kconfig
index 3b744afb8a..552df6146c 100644
--- a/drivers/mmc/Kconfig
+++ b/drivers/mmc/Kconfig
@@ -233,6 +233,14 @@ config MMC_MESON_GX
 	help
 	 Support for EMMC host controller on Meson GX ARM SoCs platform (S905)
 
+config PTBL_MBR
+	bool "Eable mbr partition format"
+	depends on MMC
+	default n
+	help
+	 This enables mbr partitioning method in the MMC.
+	 Default is N..
+
 config MMC_MXC
 	bool "Freescale i.MX21/27/31 or MPC512x Multimedia Card support"
 	help
diff --git a/drivers/mmc/aml_emmc_partition.c b/drivers/mmc/aml_emmc_partition.c
index bb570c92c5..282d073e99 100644
--- a/drivers/mmc/aml_emmc_partition.c
+++ b/drivers/mmc/aml_emmc_partition.c
@@ -23,7 +23,6 @@
 
 DECLARE_GLOBAL_DATA_PTR;
 /* using mbr*/
-#define CONFIG_PTBL_MBR	(0)
 #if (CONFIG_PTBL_MBR)
 	/* cmpare partition name? */
 	#define CONFIG_CMP_PARTNAME	(0)
-- 
2.25.1

